### PR TITLE
add recipe for essm

### PR DESCRIPTION
**Pull request by schymans**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by schymans** [comment]
The builds fail because conda cannot install pytest-runner, which has been deprecated. Is there an easy way around this, or does the testing need to be fixed in the source code of essm?

File: recipes/essm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.0
+    - pip
+    - setuptools-scm >=3.1.0
+  run:
+    - python >=3.0
+    - sympy ==1.3
```
Comment (scopatz) [inline recipes/essm/meta.yaml:24]: Does this need to be so tight of a pin?

Comment (schymans) [reply recipes/essm/meta.yaml:24]: Unfortunately yes, for the current version. There is a [PR](https://github.com/environmentalscience/essm/pull/90) waiting in essm to relax the sympy pinning, which will become the next version. Once we include one version in conda-forge, will it update automatically, or will I have to re-do the recipe for each update where pinning changes?

Comment (schymans) [reply recipes/essm/meta.yaml:24]: It uses features introduced in or just before sympy 1.3. I will update the PR to use the most recent version of sympy, which should work with sympy>=1.6.0

Comment (scopatz) [reply recipes/essm/meta.yaml:24]: So it should be <=1.3 then?

Comment (schymans) [reply recipes/essm/meta.yaml]: Thanks for looking into this. I just updated to the new version of essm with clearer dependencies. Hope this solves the issue about the sympy version, but the pytest-runner problem will probably prevail.

File: recipes/essm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  run:
+    - python >=3.0
+    - sympy >=1.6.0
+
+test:
+  imports:
+    - check-manifest>=0.25
+    - coverage >=4.0
+    - isort >=5.0.0
+    - pydocstyle >=1.0.0, <4.0.0
```
Comment (scopatz) [inline recipes/essm/meta.yaml:31]: should be `>=1.0.0, <4.0.0`

File: recipes/essm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - python >=3.0
+    - pip
+    - setuptools-scm >=3.1.0
+  run:
+    - python >=3.0
+    - sympy >=1.6.0
+
+test:
+  imports:
+    - check-manifest>=0.25
```
Comment (scopatz) [inline recipes/essm/meta.yaml:28]: Actually none of these import tests should have version constraints

File: recipes/essm/meta.yaml (hunk 2 above)
Comment (schymans) [reply recipes/essm/meta.yaml]: I thought there should be no version constraints?

Comment (schymans) [reply recipes/essm/meta.yaml]: Which one do you mean? pydocstyle was already >=1.0.0, <4.0.0

**Pull request closed without merging**